core/includes/icon.inc

```diff
@@ -0,0 +1,250 @@
+  }
+}
+
+/**
+ * Checks if core provides an icon.
+ *
+ * Do not call this function directly. Use icon() instead.
+ *
+ * @param string $icon_name
+ *   The icon name to be located.
+ *
+ * @return string|void
+ *   The path to the icon if found.
+ *
+ * @see icon()
+ * @private
+ */
+function _icon_from_core($icon_name) {
+  // Finally, check for an icon in the misc/icons directory.
+  $misc_icon_path = 'core/misc/icons/' . $icon_name . '.svg';
+  if (file_exists($misc_icon_path)) {
+    return $misc_icon_path;
+  }
+}
+
+/**
+ * Get info about a module-provided icon, or all icons.
+ *
+ * @param string|null $icon_name
+ *   An icon name, with no extension, such as "home" or "info". If omitted,
+ *   all icon info will be returned.
+ * @return array
+ */
+function icon_get_info($icon_name = NULL) {
+  $icon_info = &backdrop_static(__FUNCTION__);
+  if (!isset($icon_info)) {
+    $icon_info = module_invoke_all('icon_info');
+    backdrop_alter('icon_info', $icon_info);
+  }
+  if ($icon_name) {
+    return isset($icon_info['name']) ? $icon_info['name'] : NULL;
+  }
+  else {
+    return $icon_info;
+  }
+}
+
+/**
+ * Returns HTML for an inline-icon.
+ *
+ * This effectively returns the contents of an SVG file. But it could
+ * potentially be override to replace inlined SVGs with other mechanisms, like
+ * an icon font.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - name: The machine name for the icon being displayed.
+ *   - path: The full path to the icon file, relative to installation root.
+ *   - attributes: Attributes to be added to the icon itself.
+ *   - wrapper_attributes: An array of attributes to add to the icon wrapper.
+ */
+function theme_icon(array $variables) {
+  // Ensure the filename is .svg.
+  if (file_get_extension($variables['path']) === 'svg') {
+    // Ensure the file contents are an SVG.
+    $svg_contents = file_get_contents($variables['path']);
+    if (strpos($svg_contents, '<svg') === 0) {
+      // Clean out any embedded XSS within the SVG.
+      $svg_contents = filter_xss($svg_contents, array('svg', 'use', 'title',
+        'desc', 'defs', 'linearGradient', 'stop', 'rect', 'circle', 'path'));
+      $svg_contents = icon_add_attributes($svg_contents, $variables['attributes']);
+      $wrapper_attributes = $variables['wrapper_attributes'];
+      $wrapper_attributes['class'] = array_merge(
+        array('icon', 'icon--' . $variables['name']),
+        (array) $wrapper_attributes['class']
+      );
+      return '<span' . backdrop_attributes($wrapper_attributes) . '>' . $svg_contents . '</span>';
+    }
+  }
+}
+
+/**
+ * Add attributes to an SVG string.
+ *
+ * @param string $svg
+ *   The entire contents of an SVG image.
+ * @param array $attributes
+ *   An array of attributes to add to the SVG image.
+ *
+ * @return string
+ *   The SVG image contents with the attributes added.
+ */
+function icon_add_attributes($svg, array $attributes) {
+  // @todo: Add attributes and alt.
```

Comment:
I think, it's totally possible to just add `aria-hidden="true"` as the icons are always just part of a label (like menu items) - is that correct?

There's a thread in our Zulip chat to discuss this more deeply, if necessary https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/Inline.20SVG.20and.20A11Y